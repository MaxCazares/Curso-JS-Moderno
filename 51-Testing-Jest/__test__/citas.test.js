import Citas from "../js/classes/Citas";

describe('Probar la clase de citas', () => {

    const citas = new Citas();
    const id = Date.now();

    test('Agregar una nueva cita', () =>{
        const citaObj = {
            id,
            mascota: 'Poppy',
            propietario: 'Valeria',
            telefono: '1234567890',
            fecha: '10-12-2022',
            hora:'10:20',
            sintomas: 'solo duerme'
        }

        citas.agregarCita(citaObj);

        expect(citas).toMatchSnapshot();

    });

    test('Actualizar cita', () =>{
        const citaActualizada = {
            id,
            mascota: 'Otra mascota',
            propietario: 'Valeria',
            telefono: '1234567890',
            fecha: '10-12-2022',
            hora:'10:20',
            sintomas: 'solo duerme'
        }

        citas.editarCita(citaActualizada);
        expect(citas).toMatchSnapshot();
    });

    test('Eliminar cita', () => {
        citas.eliminarCita(id);

        expect(citas).toMatchSnapshot();
    });

});