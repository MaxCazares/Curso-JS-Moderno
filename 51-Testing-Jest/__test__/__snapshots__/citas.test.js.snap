// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase de citas Actualizar cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "10-12-2022",
      "hora": "10:20",
      "id": 1642292271088,
      "mascota": "Otra mascota",
      "propietario": "Valeria",
      "sintomas": "solo duerme",
      "telefono": "1234567890",
    },
  ],
}
`;

exports[`Probar la clase de citas Agregar una nueva cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "10-12-2022",
      "hora": "10:20",
      "id": 1642292271088,
      "mascota": "Poppy",
      "propietario": "Valeria",
      "sintomas": "solo duerme",
      "telefono": "1234567890",
    },
  ],
}
`;

exports[`Probar la clase de citas Eliminar cita 1`] = `
Citas {
  "citas": Array [],
}
`;
